history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
```

Comment:
Yeah, that's probably not the right term since the history is actually stored in the linked list of `SignedTreeRoot`s.

Sorry, I'm kind of spinning on the API we want to expose as this all becomes more familiar to me.

Is the right name `PAD` then? (being a persistent authenticated dictionary, which allows queries against older versions of the structure, to quote that paper).  The developer would instantiate a new `PAD`, which would in turn construct the Merkle tree as part of its internals.